modules/Microsoft.Sql/servers/.test/secondary/deploy.test.bicep

```diff
@@ -0,0 +1,64 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.sql.servers-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'sqlsec'
+
+@description('Optional. The password to leverage for the login.')
+@secure()
+param password string = newGuid()
+
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
+  params: {
+    serverName: '<<namePrefix>>-${serviceShort}-pri'
+    databaseName: 'db1'
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>-${serviceShort}-sec'
+    administratorLogin: 'adminUserName'
+    administratorLoginPassword: password
+    databases: [
+      {
+        name: nestedDependencies.outputs.databaseName
+        skuTier: 'Basic'
+        skuName: 'Basic'
+        maxSizeBytes: 2147483648
+        createMode: 'Secondary'
+        sourceDatabaseId: nestedDependencies.outputs.databaseId
```

Comment:
```suggestion
        databaseResourceId: nestedDependencies.outputs.databaseResourceId
```

Reply:
Changing this to: sourceDatabaseResourceId: nestedDependencies.outputs.databaseResourceId